### PR TITLE
Home w34

### PR DESCRIPTION
**Pull request by KostiantynVorobiov**

### CODE REVIEW

**Review by frostpv — APPROVED** [review]

File: src/main/resourses/intit_db.sql [hunk 1]
```diff
@@ -20,6 +20,9 @@ CREATE TABLE `internet_shop`.`users`
     UNIQUE INDEX `login_UNIQUE` (`login` ASC) VISIBLE
 );
 
+ALTER TABLE `internet_shop`.`users`
+    ADD COLUMN `salt` VARBINARY(16) NOT NULL AFTER `password`;
+
```
Comment (dmytro-hryhoruk) [inline src/main/resourses/intit_db.sql]: let's change the original alter statement

Comment (KostiantynVorobiov) [reply src/main/resourses/intit_db.sql]: Done

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 2]
```diff
@@ -15,7 +16,8 @@
     public User login(String login, String password) throws AuthenticationException {
         User userFromDb = userService.findByLogin(login).orElseThrow(() ->
                 new AuthenticationException("This user does not exist or incorrect data entered"));
-        if (userFromDb.getPassword().equals(password)) {
+        if (userFromDb.getPassword().equals(HashUtil
+                .hashPassword(password, userFromDb.getSalt()))) {
             return userFromDb;
         }
         throw new AuthenticationException("Incorrect login or password");
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: let's modify this method using isPresent() in order to throw exception only once( the second one)

Comment (KostiantynVorobiov) [reply src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: Done

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**